internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -499,6 +513,35 @@ func (r *BMACReconciler) reconcileNodeLabels(bmh *bmh_v1alpha1.BareMetalHost, ag
 	return false
 }
 
+func (r *BMACReconciler) reconcileClusterReference(bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent, infraEnv *aiv1beta1.InfraEnv) (bool, error) {
+	if clusterReferenceStr, ok := bmh.Annotations[BMH_CLUSTER_REFERENCE]; ok {
```

Comment:
It would have less indentation and easier readability if you reversed the if
```suggestion
	if clusterReferenceStr, ok := bmh.Annotations[BMH_CLUSTER_REFERENCE]; !ok {
		return false, nil
	}
```

Reply:
OK